src/controllers/helpers.js

```diff
@@ -0,0 +1,126 @@
+const { Cart } = require("../models/cart.model");
+const { Wishlist } = require("../models/wishlist.model");
+const { Product } = require("../models/product.model");
+const { User } = require("../models/user.model");
+const { HttpError } = require("../utils/helper");
+
+const findCartByUserId = async (userId) => {
+  let userCart = await Cart.findOne({ user: userId }).populate("items.product");
+  if (userCart) {
+    userCart.calculateTotalPrice();
+    userCart = await userCart.save();
```

Comment:
I guess assignment is unnessary `userCart = await userCart.save()`.
`await userCart.save()` will mutate `userCart` object.

Reply:
Resolved.